### PR TITLE
function: fix grouping function result name

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/7717

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
use tiflash pr: https://github.com/pingcap/tiflash/pull/7169
```
create table t(a int, b int, c int);
alter table t set tiflash replica 1;
insert into t values(1,1),(1,2);
select count(1), grouping(a), grouping(b) from t group by a,b with rollup;
select count(1), grouping(b), grouping(a) from t group by a,b with rollup;
```
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
function: fix grouping function result name
```

### CODE REVIEW

**Review by xzhangxian1008 — APPROVED** [review]
LGTM

File: dbms/src/Flash/Coprocessor/DAGExpressionAnalyzerHelper.cpp [hunk 1]
```diff
@@ -415,7 +415,12 @@ String DAGExpressionAnalyzerHelper::buildGroupingFunction(
         argument_names.push_back(name);
     }
 
+    // Get the result by function name and parameters is **NOT** enough, grouping functions like: grouping(a) and grouping(b) will be rewritten as the same signature:
+    //  grouping(gid), grouping(gid) with different metadata.
+    // In TiDB, we don't reuse the action by the signature name, while TiFlash does, we should distinguish different grouping function out from their metadata.
     String result_name = genFuncString(func_name, argument_names, {getCollatorFromExpr(expr)});
+    // grouping function's metadata has naturally been encoded as proto-message as string, just appending them to the result name as new grouping functions' result_name.
+    result_name += expr.val();
```
Comment (SeaRise) [inline dbms/src/Flash/Coprocessor/DAGExpressionAnalyzerHelper.cpp:423]: Will this prevent other functions from reusing the results?

**Review by SeaRise — APPROVED** [review]
lgtm

**Comment by SeaRise** [comment]
@AilinKid 
You can comment `/hold cancel` when you want to merge this pr

**Pull request merged**